File: frontend/src/lib/api/tenancy/types/location-branding.ts
import { z } from 'zod';

export const publicLocationResponseSchema = z.object({
    tenant_name: z.string(),
    tenant_code: z.string(),

    location_code: z.string(),
    location_name: z.string(),

    display_name: z.string().nullable(),
    logo_key: z.string().nullable(),

    primary_color: z.string().nullable(),
    secondary_color: z.string().nullable(),
    font_family: z.string().nullable()
});

export type PublicLocationResponse =
    z.infer<typeof publicLocationResponseSchema>;

File: backend/app/tenancy/schemas/location_branding.py
from pydantic import BaseModel


class PublicLocationResponse(BaseModel):
    tenant_name: str
    tenant_code: str

    location_code: str
    location_name: str

    display_name: str | None
    logo_key: str | None

    primary_color: str | None
    secondary_color: str | None
    font_family: str | None